sdk/common/src/main/scala/org/typelevel/otel4s/sdk/Attributes.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk
+
+import cats.Show
+import cats.implicits.showInterpolator
+import org.typelevel.otel4s.Attribute
+import org.typelevel.otel4s.Attribute.KeySelect
+import org.typelevel.otel4s.AttributeKey
+
+/** An immutable collection of [[Attribute]]s.
+  */
+final class Attributes private (
```

Comment:
Currently, most methods use varargs, e.g:
```scala
def resourceSpan[A](name: String, attributes: Attribute[_]*)
```

I wonder whether we should offer a new alternative too:
```scala
def resourceSpan[A](name: String, attributes: Attributes)
```

Reply:
To be honest, I would completely replace Attribute with Attributes. Attribute looks like an unnecessary wrapper to me because, from my perspective, users are more interested in using a collection that provides interfaces for accessing particular values by keys or iterating over all of them. I can imagine that It might be useful as a smart constructor which validates passed values, but the collection can do the same. Maybe I don't see a use-case for it.